### PR TITLE
Integrating Scilla Server

### PR DESCRIPTION
**Pull request by kaikawaliu**
## Description

Implementation reference:
https://github.com/Zilliqa/scilla/wiki/Scilla-Server-API 

Refactored some functions in AccountStoreSC.tpp

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libData/AccountData/AccountStoreSC.tpp [hunk 1]
```diff
@@ -315,51 +323,12 @@ bool AccountStoreSC<MAP>::UpdateAccounts(const uint64_t& blockNum,
...
-          };
-          DetachedFunction(1, func2);
-
-          {
-            std::unique_lock<std::mutex> lk(m_MutexCVCallContract);
-            cv_callContract.wait(lk);
-          }
+          // invoke scilla runner
+          InvokeInterpreter(RUNNER_CREATE, runnerPrint, scilla_version,
+                            is_library, gasRemained, amount, ret, receipt);
```
Comment (ansnunez) [inline src/libData/AccountData/AccountStoreSC.tpp:328]: It seems like in the original code we called runner with balance 0, whereas now we're using `amount` for the balance. Is this a code fix?

File: src/libData/AccountData/ScillaClient.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2019 Zilliqa
```
Comment (ansnunez) [inline src/libData/AccountData/ScillaClient.cpp:2]:
```suggestion
 * Copyright (C) 2020 Zilliqa
```

File: src/libData/AccountData/ScillaClient.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 Zilliqa
```
Comment (ansnunez) [inline src/libData/AccountData/ScillaClient.h:2]:
```suggestion
 * Copyright (C) 2020 Zilliqa
```

File: src/libUtils/ScillaUtils.cpp [hunk 4]
```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2019 Zilliqa
```
Comment (ansnunez) [inline src/libUtils/ScillaUtils.cpp:2]:
```suggestion
 * Copyright (C) 2020 Zilliqa
```

File: src/libUtils/ScillaUtils.h [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2019 Zilliqa
```
Comment (ansnunez) [inline src/libUtils/ScillaUtils.h:2]:
```suggestion
 * Copyright (C) 2020 Zilliqa
```

File: constants.xml [hunk 6]
```diff
@@ -156,6 +156,8 @@
         <IP_TO_BIND>127.0.0.1</IP_TO_BIND>
         <ENABLE_STATUS_RPC>true</ENABLE_STATUS_RPC>
         <SCILLA_IPC_SOCKET_PATH>/tmp/zilliqa.sock</SCILLA_IPC_SOCKET_PATH>
+        <SCILLA_SERVER_SOCKET_PATH>/tmp/scilla-server.sock</SCILLA_SERVER_SOCKET_PATH>
+        <SCILLA_SERVER_BINARY>scilla-server</SCILLA_SERVER_BINARY>
```
[the remark this answers is not in the document]
Comment (kaikawaliu) [reply constants.xml:160]: it'll be created within the c++ code, so won't be an issue during set up

File: src/libData/AccountData/AccountStoreSC.tpp (hunk 1 above)
Comment (kaikawaliu) [reply src/libData/AccountData/AccountStoreSC.tpp:328]: yes you can say that, but during deployment the amount in the command doesn't really matter, so we were fine in the past

**Pull request merged**